zulipterminal/core.py

```diff
@@ -182,8 +188,7 @@ def stream_muting_confirmation_popup(self, button: Any) -> None:
                               "center")
         mute_this_stream = partial(self.model.toggle_stream_muted_status,
                                    button.stream_id)
-        self.loop.widget = PopUpConfirmationView(self, question,
-                                                 mute_this_stream)
+        self.show_confirmation_overlay(question, mute_this_stream)
```

Comment:
I'm not sure that we really gain from this refactor?

I think we could probably benefit from combining the confirmation popups into 'show_pop_up', however, though that doesn't seem necessary to include here but is pretty independent.

I know you use it in the next commit, but generally I'd prefer if we kept helper code clean of controller/UI, and have the latter use the helper code instead.